rust-runtime/aws-smithy-json/src/deserialize/error.rs

```diff
@@ -82,3 +82,12 @@ impl From<EscapeError> for Error {
         }
     }
 }
+
+impl From<aws_smithy_types::TryFromNumberError> for Error {
+    fn from(_: aws_smithy_types::TryFromNumberError) -> Self {
+        Error {
+            reason: ErrorReason::InvalidNumber,
+            offset: None,
```

Comment:
We don't have the offset here. We only have the offset when working with the token directly in the code-generated deserializer, or when calling the `expect_` functions which auto-fill it from the token in case of errors. When unescaping strings, we're also bubbling up errors without offsets. Making both cases bubble up offsets would require some refactoring.